service-front/module/Application/src/Controller/DonorPostOfficeFlowController.php

```diff
@@ -82,105 +82,61 @@ public function doDetailsMatchAction(): ViewModel
         return $view->setTemplate('application/pages/post_office/donor_details_match_check');
     }
 
-    public function findPostOfficeAction(): ViewModel|Response
-    {
-        $view = new ViewModel();
-        $uuid = $this->params()->fromRoute("uuid");
-
-        $form = (new AttributeBuilder())->createForm(PostOfficePostcode::class);
-        $view->setVariable('form', $form);
-
-        $detailsData = $this->opgApiService->getDetailsData($uuid);
-        $optionsdata = $this->config['opg_settings']['post_office_identity_methods'];
-
-        foreach ($detailsData['address'] as $line) {
-            if (preg_match('/^[A-Z]{1,2}[0-9]{1,2}[A-Z]? [0-9][A-Z]{2}$/', $line)) {
-                $defaultPostcode = $line;
-            }
-        }
-
-        if (count($this->getRequest()->getPost())) {
-            $formObject = $this->getRequest()->getPost();
-            $formData = $formObject->toArray();
-            if ($formData['postcode'] == 'alt') {
-                $postcode = $formData['alt_postcode'];
-            } else {
-                $postcode = $formData['postcode'];
-            }
-            $formObject->set('selected_postcode', $postcode);
-            $form->setData($formObject);
-
-            if ($form->isValid()) {
-                $response = $this->opgApiService->addSearchPostcode($uuid, $postcode);
-                if ($response['result'] === 'Updated') {
-                    return $this->redirect()->toRoute('root/find_post_office_branch', ['uuid' => $uuid]);
-                }
-            }
-        }
-
-        /**
-         * @psalm-suppress PossiblyUndefinedVariable
-         */
-        $view->setVariable('default_postcode', $defaultPostcode);
-        $view->setVariable('options_data', $optionsdata);
-        $view->setVariable('details_data', $detailsData);
-        $view->setVariable('uuid', $uuid);
-
-        return $view->setTemplate('application/pages/post_office/find_post_office');
-    }
-
     public function findPostOfficeBranchAction(): ViewModel|Response
     {
+        $templates = ['default' => 'application/pages/post_office/find_post_office_branch'];
         $view = new ViewModel();
         $uuid = $this->params()->fromRoute("uuid");
 
         $optionsdata = $this->config['opg_settings']['post_office_identity_methods'];
```

Comment:
Do we need `optionsdata` in the postoffice lookup page? I couldn't see it used in the twig.